### ISSUE TITLE
Allow configurable prefix in DefinePlugin

### ISSUE THREAD

**Issue by k88hudson** [webpack/webpack#2129]
I noticed that `DefinePlugin` already has code to support a prefix, but it's hardcoded to `""` at the moment. Would you be open to a pull request to enable it as a second param or as part of an options object?

**Comment by bebraw** [comment]
Go ahead and PR.

What's the use case for this?

**Comment by neeharv** [comment]
@bebraw Should I add support for this as part of #3632 ?

**Comment by bebraw** [comment]
@neeharv It would be better as a separate PR as it's easier to review then. I guess #3632 has to be merged before that, though.

**Comment by scottdj92** [comment]
So, I tried this out. Introducing an options parameter for prefixes causes a lot of stuff to break down the line. I'm trying to think of a better solution than this one:

```javascript
			params.normalModuleFactory.plugin("parser", (parser) => {
				(function walkDefinitions(definitions, prefixes) {
					Object.keys(definitions).forEach((key) => {
						let code = definitions[key];
						if(code && typeof code === "object" && !(code instanceof RegExp)) {
							if(prefixes !== null || prefixes !== undefined) {
								Object.keys(prefixes).forEach((prefix) => {
									walkDefinitions(code, prefix + key + ".");
									applyObjectDefine(prefix + key, code);
								});
							} else {
								walkDefinitions(code, "" + key + ".");
								applyObjectDefine("" + key, code);
							}
							return;
						}
						applyDefineKey(prefixes, key);
						applyDefine(prefixes + key, code);
					});
				}(definitions, prefixes));
```

**Comment by jaredramirez** [comment]
Hey! If this is still open, i'd like to help out with it as my first contribution to webpack.

**Comment by bebraw** [comment]
@jaredramirez Sure, go ahead. Note Scott's observation above, though.

**Comment by Madhu94** [comment]
@jaredramirez Are you working on this ?

**Comment by jaredramirez** [comment]
I had forgotten about this! I will take another look at this this week/weekend, I'm sorry for the delay.

**Comment by AKST** [comment]
Would it be worth closing this, since @sokra & @ooflorent have suggested in #7730 it's not really needed? I'm only suggesting this as it appears @SerhiiBilyk went to the effort of writing a PR & applying PR feedback only to be told it's not needed 😕

**Comment by SerhiiBilyk** [comment]
@AKST thanks. Don't know why they don't close this issue if it is not needed.
